### ISSUE TITLE
codecov/project fails on every PR: coverage uploads from main have been rejected since June 2024

### ISSUE THREAD

**Issue by Eljees** [stackrox/kube-linter#1242]
`codecov/project` is red on every open pull request that has a codecov context - #1189, #1197, #1206, #1216, #1236 - and green on none of them. The cause is not in those branches.

## Codecov has had no report for `main` since June 2024

`api.codecov.io/api/v2/github/stackrox/repos/kube-linter/commits/?branch=main` returns 19 commits, the newest being `dbd7529` - "chore(deps): bump helm.sh/helm/v3 from 3.15.1 to 3.15.2 (#796)", 13 June 2024.

Every PR is therefore compared against a base 340+ commits behind, and two years of intervening work counts as the patch. On #1236 codecov reports 1,959 patch lines against the 254 that branch actually adds, and a base of 197 files against a head of 239.

## The uploads are being rejected, silently

From the last `push`-triggered run on `main` ([run 28674157419](https://github.com/stackrox/kube-linter/actions/runs/28674157419), 3 July 2026, `build-and-test`):

```
info  - 2026-07-03 17:18:30,985 -- Found 1 coverage files to report
info  - 2026-07-03 17:18:30,985 -- > /home/runner/work/kube-linter/kube-linter/coverage.out
info  - 2026-07-03 17:18:31,615 -- Upload queued for processing complete
error - 2026-07-03 17:18:31,615 -- Upload queued for processing failed: {"message":"Repository not found"}
```

The step still reports success, because `fail_ci_if_error` defaults to false. That is why this has gone unnoticed for so long.

"Repository not found" is what codecov returns for an upload token that does not match the repository, and the rest of the evidence fits: the branch list on codecov contains only fork branches - `pujitha24:...`, `ugiordan:...`, `mwozniakk:...`, `bdsoha:...`, `arpitjain099:...` - and no in-repo branch at all. Pull requests from forks have no access to `secrets.CODECOV_TOKEN`, so they upload tokenless and succeed; the runs that do have the secret send a token codecov rejects.

## A second gap, independent of the token

`.github/workflows/build.yaml` has not run on a `push` to `main` since 3 July, yet `main` has had 12 commits since. All of them are dependabot merges, and `.github/workflows/auto-merge.yaml` performs those with

```yaml
run: gh pr merge --auto --squash "$PR_URL" && gh pr review --approve "$PR_URL"
env:
  GH_TOKEN: ${{secrets.GITHUB_TOKEN}}
```

Pushes made with `GITHUB_TOKEN` do not trigger further workflow runs, so those merges produce no build and no coverage upload. Even with a working token, `main` would stay dark for as long as its history is dependabot auto-merges.

## What would fix it

1. Rotate `CODECOV_TOKEN` to the current upload token for `stackrox/kube-linter`. That needs repository settings access.
2. Set `fail_ci_if_error: true` on both upload steps in `build.yaml`, so the next silent rejection is visible rather than costing two years.
3. For the second gap, either merge dependabot PRs with an app token or a PAT, or add a `schedule`/`workflow_dispatch` trigger to `build.yaml` so `main` gets a periodic coverage run.

I am happy to send 2 and 3 as a PR if that is useful - 1 is not something I can do from outside.
